src/UI/Actions.hs

```diff
@@ -665,7 +666,7 @@ findIndexWithOffset i fx = fmap (i+) . Vector.findIndex fx . Vector.drop i
 
 listSetSelectionEnd :: Lens' AppState (L.List Name a) -> AppState -> AppState
 listSetSelectionEnd list s =
-  let index = Vector.length $ view (list . L.listElementsL) s
+  let index = lengthOf (list . L.listElementsL) s
```

Comment:
hm.. TIL

Reply:
I worked out the problem; I needed a ` . folded` at the end.  `lengthOf` counts the number of
targets of the optic.  In case of a *lens* it is always ``1``.  But actually `lengthOf` is not efficient *O(n)*.
It's better to `. to length` as the last component in the optic, which will use the `Foldable Vector`
instance for *O(1)* complexity.  Updated commit imminent.